shared/js/media/media_frame.js

```diff
@@ -245,7 +247,20 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
     }
   }
   else {
-    noPreview();
+    // If we have valid small EXIF preview use it for css background image
+    // This will help to load large downsampled preview images without flash
+    // See bug 1188286
+    // ( TBD: preview & preview.start might not be enough.
+    // Need more robust check for small valid EXIF preview )
+    if (preview && preview.start) {
+      // Create a blob URL for the bg preview
+      self.bgPreviewURL = URL.createObjectURL(
```

Comment:
Please initialize this.bgPreviewURL to null up at the start of displayImage() where we initialize other things that later get cleared. Maybe after this.fullImageURL. Then add a comment there explaining what the field is for.